src/sentry/integrations/github/client.py

```diff
@@ -164,15 +209,19 @@ def create_token(self):
         )
 
     def check_file(self, repo: Repository, path: str, version: str) -> str | None:
```

Comment:
Why does this function have a return type of `str | None`? `file` is not optional.

Reply:
Ok this is weird so I'll leave a comment in the file. _GitLab's_ `check_file()` returns an optional string so in order for `RepositoryMixin` to type-check they're the same method signature.